### DISCUSSION TITLE
How can I print a model to the screen?

### DISCUSSION THREAD

**Discussion by erelsgl** [coin-or/python-mip#336] [Q&A] [answered]
I created a model in code, and the solution is wrong. To debug the issue, I would like to print the model to the screen. When I write `print(model)`. I get `MIP model: <mip.model.Model object at 0x00000192D35C6950>`.

Is there a way to print the actual model (the constraints and objective) to the screen?

**Comment by rschwarz** [comment, marked as answer]
You can use `Model.write("problem.lp")` to write the problem to a file in the `.lp` format, which is human-readable.

**Comment by erelsgl** [comment]
OK. This helps, but I think it would be better to be able to get the model as a string. This would allow e.g. to write the model to a log file using the standard `logging`, print it to the console, etc.

**Reply by sebheger** [reply]
@erelsgl The best way at the moment to do that is already stated by @rschwarz 
There might be a useful extension to write the model into a filestream to work with it. There was already a lot of discussion, but we most likely will not support something like print(model) to print all associated model entities.

**Reply by rschwarz** [reply]
Note that you can iterate over the `model.constrs` and call `str(c)` on each, to get a representation that way.
